### ISSUE TITLE
How to filter for test tasks

### ISSUE THREAD

**Issue by amitlisha** [samacqua/LARC#4]
Hey,
In the paper you mentioned splitting the dataset to 200 training tasks, and 183 test tasks.
I can't find any field indicating the split, how can I reproduce it?

Thanks!

EDIT:
Ok I found the datasheet, you guys explain there that you took the first 200 for training, and the rest for test.
I filtered the other half of samples by iterating each description and each build, and leaving out samples with only failed builds. I was left with 181 where I should've been left with 183 - as mentioned in the paper.
I also tried filtering the first 200 - the train, and I was left with 183, so maybe you used the first 200 samples for the test split?

I ordered them by the json name in `dataset/tasks_json/` - 0 to 399.

**Comment by samacqua** [comment]
Hey! Sorry for the late response and confusing train/test split.

The data sheet is actually a bit misleading. The train/test split is now in `dataset/summary/train_test_split.json`, it is generated from the first/last 200 tasks from  the task IDs sorted alphabetically, then a random shuffle with seed 0. See below script for verification:

```python
"""Verify the train/test split used in the paper.

Shows that:
1. The split is reproducible via random.shuffle with seed=0 on sorted ARC task filenames.
2. 183 of the 200 test tasks have language annotations (successfully communicated
   descriptions used in the program synthesis experiments).

The split was defined in the program synthesis codebase:
  https://github.com/theosech/ec/tree/language-guided_program_synthesis_for_larc
in data/larc/train_test_split.json (generated by dreamcoder/domains/arc/main.py's
train_test_split function: random.seed(0), shuffle sorted ARC filenames, 50/50 split).

The language annotations (data/arc/language/sentences/language.json) contain 354 of the
400 tasks — those with successfully communicated natural programs. Of the 200 test tasks,
183 have annotations. This is the "183 test tasks" reported in the paper.

Note: the datasheet says "first 200 as train and rest 200 as test" which is a simplified
description. The actual split is a seeded random shuffle, not ordered by LARC task ID.
"""
import json
import random

with open("dataset/train_test_split.json") as f:
    split = json.load(f)

# --- 1. Reproduce the split from random.seed(0) on sorted ARC filenames ---

task_names = sorted(split["train"] + split["test"])
assert len(task_names) == 400

random.seed(0)
random.shuffle(task_names)
generated_train = task_names[:200]
generated_test = task_names[200:]

assert generated_train == split["train"]
assert generated_test == split["test"]
print("1. random.seed(0) + shuffle on sorted ARC filenames reproduces the split exactly")

# --- 2. 183/200 test tasks have language annotations ---

# language.json contains processed NL annotations for the 354 tasks with successfully
# communicated descriptions (from the program synthesis repo's data pipeline).
with open(
    "language-guided_program_synthesis_for_larc/data/arc/language/sentences/language.json"
) as f:
    language_tasks = set(json.load(f).keys())

test_with_lang = [t for t in split["test"] if t in language_tasks]
train_with_lang = [t for t in split["train"] if t in language_tasks]

print(f"2. Test tasks with language annotations:  {len(test_with_lang)}/200")
print(f"   Train tasks with language annotations: {len(train_with_lang)}/200")
print(f"   Total tasks with annotations:          {len(language_tasks)}/400")

assert len(test_with_lang) == 183
assert len(train_with_lang) == 171
assert len(language_tasks) == 354
print("   Matches the 183 test tasks reported in the paper")

```

**Comment by amitlisha** [comment]
thanks!
